core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -368,6 +418,21 @@ object LAFuture {
     result
   }
 
+
+  /**
+   * Collect all the future values into the aggregate future
+   * The returned future will be satisfied when all the
+   * collected futures are satisfied
+   */
+  def collect[T](future: LAFuture[T]*): LAFuture[List[T]] = {
+    collect[T, List[T]](
+      { (value, result, values, index) => values.insert(index, Full(value)) },
+      { (valueBox, result, values, index) => result.fail(valueBox) },
+      { (result, values) => result.satisfy(values.toList.flatten) },
+      future: _*
```

Comment:
For clarity, I think we should either...

1. Use the argument names in the method invocation (so, `valueSucceeded = { (value result...`
2. Declare these callbacks as their own `def`s within `collect[T]` and pass them into the collect call by name

I like option (2) the best, but could be convinced either is better. =)